src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/OrbitMetricsInvocationHandler.java

```diff
@@ -37,53 +38,60 @@
 import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Created by jgong on 12/19/16.
  */
 public class OrbitMetricsInvocationHandler extends DefaultInvocationHandler
 {
-    private Map<String, Histogram> actorMethodResponseTimeHistograms = new ConcurrentHashMap<>();
-    private Map<String, Histogram> actorChainResponseTimeHistograms = new ConcurrentHashMap<>();
+    private Map<String, Timer> actorMethodResponseTimeTimers = new ConcurrentHashMap<>();
+    private Map<String, Timer> actorChainResponseTimeTimers = new ConcurrentHashMap<>();
 
     @Override
-    public void afterInvoke(final long startTimeNanos, final Invocation invocation, final Method method) {
+    public void afterInvoke(final long startTimeNanos, final Invocation invocation, final Method method)
+    {
         final RemoteReference toReference = invocation.getToReference();
         super.afterInvoke(startTimeNanos, invocation, method);
         final long durationNanos = (System.nanoTime() - startTimeNanos);
         final Double durationMs = durationNanos / 1_000_000.0;
         Class actorClass = RemoteReference.getInterfaceClass(toReference);
-        String histKey = getActorMethodResponseTimeMetricKey(actorClass, method.getName());
-        Histogram hist = actorMethodResponseTimeHistograms.get(histKey);
-        if (null == hist) {
-            hist = MetricsManager.getInstance().getRegistry().histogram(histKey);
-            actorMethodResponseTimeHistograms.put(histKey, hist);
+        String timerKey = getActorMethodResponseTimeMetricKey(actorClass, method.getName());
+        Timer timer = actorMethodResponseTimeTimers.get(timerKey);
+        if (null == timer)
+        {
+            timer = MetricsManager.getInstance().getRegistry().timer(timerKey);
+            actorMethodResponseTimeTimers.put(timerKey, timer);
         }
-        hist.update(durationMs.intValue());
+        timer.update(durationMs.longValue(), TimeUnit.MILLISECONDS);
```

Comment:
Shouldn't need to convert to millis just to send it to a method that will most likely convert it back to nanos as that is how the timer internally represents the time.